pkg/skuba/actions/node/upgrade/apply.go

```diff
@@ -96,6 +96,15 @@ func Apply(client clientset.Interface, target *deployments.Target) error {
 		if err := node.AddTargetInformationToInitConfigurationWithClusterVersion(target, initCfg, nodeVersionInfoUpdate.Update.APIServerVersion); err != nil {
 			return errors.Wrap(err, "error adding target information to init configuration")
 		}
+
+		// Upgrade 1.17 to 1.18.
+		// This updated UseHyperKube field in-memory (unsets it).
+		// Note: The cluster cm is uploaded at the end of the kubeadm process, as usual.
+		// The whole paragraph can be removed when upgrading from 1.17 is removed.
+		if currentClusterVersion.Minor() == 17 {
```

Comment:
Can't we compare latestVersion.Minor() == 18 instead?

Reply:
I don't think so. We don't need what's the latest version available. We need to know, when upgrading, that the user is at version 17. Assuming you're at version 16 when 18 is released, wouldn't that be problematic?

Reply:
My thought is that after 1.18, K8s don't need the hyperkube image.
Therefore, check the target k8s version to be upgraded is more readable for me 😄

Reply:
I expect the comment is clear enough. Thanks for the suggestion though.